packages/kbn-es-archiver/src/lib/indices/create_index_stream.ts

```diff
@@ -141,20 +141,35 @@ export function createCreateIndexStream({
           log.debug(`Deleted saved object index [${index}]`);
         }
 
+        // create the index without the aliases
         await client.indices.create(
           {
             index,
             body: {
               settings,
               mappings,
-              aliases,
             },
           },
           {
             headers: ES_CLIENT_HEADERS,
           }
         );
 
+        // create the aliases on a separate step (see https://github.com/elastic/kibana/issues/158918)
+        const actions: estypes.IndicesUpdateAliasesAction[] = Object.keys(aliases ?? {}).map(
+          (alias) => ({
+            add: {
+              index,
+              alias,
+              ...aliases![alias],
+            },
+          })
+        );
+
+        if (actions.length) {
+          await client.indices.updateAliases({ body: { actions } });
+        }
```

Comment:
afaik esArchiver will create SO indices without aliases, and then (if needed) will create aliases.

Don't we need a custom logic to make sure that recently create index is green before calling `indices.updateAliases` ?

Reply:
That's a good point, IDK if it's really necessary (as we're not querying for anything), or it it may cause an `unavailable_shards_exception` if we do not (as it happened with https://github.com/elastic/kibana/issues/157968).

Either way, I suppose it won't hurt to play it safe and wait for _green_. I'll update the PR.

Reply:
In that case we might be better off not trying to wait for green? The 50 flaky runs have passed.
Worst case scenario we'd have new flakiness 😅